tokeniser.go

```diff
@@ -146,9 +142,13 @@ func (t *Tokeniser) tokeniseNoQuotes(b byte) {
 			t.put(b)
 		}
 	}
+
+	return
 }
 
 // tokeniseSingleQuotes tokenises a single byte within single quotes.
+// It doesn't need to return whether we've finished a line, because we can't finish
+// a line in quotes.
```

Comment:
maybe they could all return a boolean value (always false in the quoted cases) - then you can replace the switch in tokeniseByte with a map?